### ISSUE TITLE
8.14.0-beta InAppMessaging on iOS: .appData is always nil even on published campaign.

### ISSUE THREAD

**Issue by abiesteves** [firebase/firebase-ios-sdk#9546]
### Step 0: Are you in the right place?

  * For issues or feature requests related to `InAppMessaging` code in this repository
    file a Github issue.
  * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: **13.0**
  * Firebase SDK version: **8.14.0-beta**
  * Installation method:  **CocoaPods**
  * Firebase Component: **In App Messaging**
  * Target platform(s):  **iOS**

### [REQUIRED] Step 2: Describe the problem

Cannot get any form of data in any delegate. InAppMessagingDisplayMessage .appData or even .campaignInfo.campaignName is all nil. I have read that it only happens on testing but even on published campaigns i couldn't get any data.

#### Steps to reproduce:

Xcode 13.0 running in a iPhone OS 15.1. 
I already have `-FIRDebugEnabled` arguments passed on launch.

#### Relevant Code:

```
// TODO(you): code here to reproduce the problem
```

**Comment by rizafran** [comment]
Thanks for reporting, @abiesteves. I tried to reproduce the issue (using 8.14.0-beta), but the value for `.appData` or `.campaignInfo` is not returning `nil`. Could you share an mcve and detailed steps to reproduce the issue?

**Comment by abiesteves** [comment]
Thank you for responding @rizafran.
**_In Code:_** In the AppDelegate, didFinishLaunchingWithOptions I called `InAppMessaging.inAppMessaging().delegate = self`. Extending the AppDelegate for the `InAppMessagingDisplayDelegate`. All delegates are called but could not get the custom data which is the .appData.

**_In the console:_** Used both Modal and Card. Added some values in the additional options. For the custom data. Tested with my Installation ID and published campaign.
>Noticed an issue in the Modal layout: the messageClicked is **never** called. (assumed should be called when user clicks on the primary button)

We've also used this same method for our cloud messaging and it works just fine.

**Comment by abiesteves** [comment]
I have tried your method @rizafran calling it in the ViewController. Upon re-running multiple times, from a published campaign. I managed to get the `.appData` but only once. I have no problem with the delegate. All are called. Just the data is always nil on my side. Can you give me steps on how you created and tested the campaigns?

Here is what I did to call the delegate in the AppDelegate.

All in the `AppDelegate.swift`
```
import Firebase
import FirebaseInAppMessaging
class AppDelegate: UIResponder, UIApplicationDelegate
    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions:   [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
        FirebaseApp.configure()
        InAppMessaging.inAppMessaging().delegate = self
    }
}

extension AppDelegate : InAppMessagingDisplayDelegate {

    func messageClicked(_ inAppMessage: InAppMessagingDisplayMessage, with action: InAppMessagingAction) {
        print(" check : messageClicked", inAppMessage.appData)
    }
}
```

**Comment by rizafran** [comment]
Thanks for sharing your code, @abiesteves. Here's what I did in the console:
1. Message layout - Modal (with button and button action)
2. Target - I just set the campaign name and app bundle ID
3. Scheduling - no end date; trigger the in-app message with `app_launch` or `on_foreground event`
4. Additional options - I added 2 custom data

Then I just tried using the code you shared in the `AppDelegate.swift`, and the `messageClicked` function was successfully being called, but the `.appData` is still returning the value of custom data from the console.

Here's what I did to test the app:
1. Upon opening the newly installed app, the pop up message will appear
2. Click the button in the pop up message
3. `inAppMessage.appData` returns the value of the custom data from the console
4. Delete and re-run the app in order for the pop up message to show again then repeat step 2-3
